### ISSUE TITLE
code to generate the dataset (python, tensorflow, keras)

### ISSUE THREAD

**Issue by saida11000** [keras-team/keras-io#411]
hello , i started to learn python, tensorflow and keras and i didn't understant the geneartion dataset code .
i need some help and comments to better understand the instructions https://keras.io/examples/vision/image_classification_from_scratch/

Generate a Dataset
image_size = (180, 180)
batch_size = 32

train_ds = tf.keras.preprocessing.image_dataset_from_directory(
"PetImages",
validation_split=0.2,
subset="training",
seed=1337,
image_size=image_size,
batch_size=batch_size,
)
val_ds = tf.keras.preprocessing.image_dataset_from_directory(
"PetImages",
validation_split=0.2,
subset="validation",
seed=1337,
image_size=image_size,
batch_size=batch_size,
)

**Comment by 8bitmp3** [comment]
Check out the API docs: https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image_dataset_from_directory

and the tutorials which use the `tf.keras.preprocessing.image_dataset_from_directory` API:
- Image classification: https://www.tensorflow.org/tutorials/images/classification
- Transfer learning and fine-tuning https://www.tensorflow.org/tutorials/images/transfer_learning
- Load images https://www.tensorflow.org/tutorials/load_data/images

If in doubt, try the tutorials on www.tensorflow.org/tutorials and keras.io

Hope this helps

**Comment by SuryanarayanaY** [comment]
Hi @saida11000 ,

Please refer the tf.keras.preprocessing.image_dataset_from_directory API for more details. This API can load the images from local directory but it should be in the form like below, i,e main directory have sub-directories for each class and under each class respective images should be there

```
main_directory/
...class_a/
......a_image_1.jpg
......a_image_2.jpg
...class_b/
......b_image_1.jpg
......b_image_2.jpg
```
If the image dataset have that structure then this API can infer the images and labels and can generate a tf.data.Dataset hat yields batches of images from the subdirectories `class_a` and `class_b`, together with labels 0 and 1 (0 corresponding to class_a and 1 corresponding to class_b).

Please refer the resources attached in above [comment](https://github.com/keras-team/keras-io/issues/411#issuecomment-801193175) for how to use this API. 

Thanks!
